go-controller/pkg/ovn/namespace.go

```diff
@@ -218,7 +236,66 @@ func (oc *Controller) updateNamespace(old, newer *kapi.Namespace) {
 	}
 	defer nsInfo.Unlock()
 
-	annotation := newer.Annotations[hotypes.HybridOverlayExternalGw]
+	var annotation, oldAnnotation string
+	annotation = newer.Annotations[routingExternalGWsAnnotation]
+	oldAnnotation = old.Annotations[routingExternalGWsAnnotation]
+	if annotation != oldAnnotation {
+		var stderr string
+		var err error
+		for podIP, gwToGr := range nsInfo.podExternalRoutes {
+			for gw, gr := range gwToGr {
+				// TODO figure out how to delete route with gw
+				_, stderr, err = util.RunOVNNbctl("--if-exists", "--policy=src-ip",
+					"lr-route-del", gr, podIP+"/32", gw)
+			}
+			if err != nil {
+				klog.Errorf("Unable to delete src-ip route to GR router, stderr:%q, err:%v", stderr, err)
+			} else {
+				delete(nsInfo.podExternalRoutes, podIP)
+			}
+		}
+		if annotation != "" {
+			annotationOk := true
+			nsInfo.routingExternalGWs = nil
+			for _, v := range strings.Split(annotation, ",") {
+				parsedAnnotation := net.ParseIP(v)
+				if parsedAnnotation == nil {
+					klog.Errorf("Could not parse routing external gw annotation")
```

Comment:
as above, include more details in the error

Reply:
done